pkg/rng/entropy.go

```diff
@@ -0,0 +1,160 @@
+package rng
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/insomniacslk/systemboot/pkg/recovery"
+)
+
+// The concept
+//
+// Most systems in real world application do not provide enough entropy at boot
+// time. Therefore we will seed /dev/random with /dev/hwrng if a HW random
+// number generator is available. Entropy is important for cryptographic
+// protocols running in network stacks. Also disk encryption can be a problem
+// if bad or no entropy is available. It can either block provisioning or makes
+// a symmetric key easy to re-calculate.
+
+var (
+	// HwRandomCurrentFile shows/sets the current
+	// HW random number generator
+	HwRandomCurrentFile = "/sys/class/misc/hw_random/rng_current"
+	// HwRandomAvailableFile shows the current available
+	// HW random number generator
+	HwRandomAvailableFile = "/sys/class/misc/hw_random/rng_available"
+	// RandomEntropyAvailableFile shows how much of the entropy poolsize is used
+	RandomEntropyAvailableFile = "/proc/sys/kernel/random/entropy_avail"
+	// EntropyFeedTime sets the loop time for seeding /dev/random by /dev/hwrng
+	// in seconds
+	EntropyFeedTime time.Duration = 2
+	// EntropyBlockSize sets the bytes to read per Read function call
+	EntropyBlockSize = 128
+	// EntropyThreshold is used to stop seeding at specific entropy level
+	EntropyThreshold uint64 = 3000
+	// RandomDevice is the linux random device
+	RandomDevice = "/dev/random"
+	// HwRandomDevice is the linux hw random device
+	HwRandomDevice = "/dev/hwrng"
+)
+
+// Can be extended but keep in mind to priorize
+// more secure random sources like hw random over
+// timer, jitter based mechanisms. At the top of the array
+// is the highest priority.
+// <rng-name>
+var trngList = []string{
+	"tpm-rng",
+	"intel-rng",
+	"amd-rng",
+	"timeriomem-rng",
+}
+
+// Searches for available True Random Number Generator
```

Comment:
In Go there is a preference for starting a comment with "$name" or "The $name" (where $name is the function/struct/whatever name). This is not enforced for non-public objects but it's still a good idea to follow this

Reply:
Done